### ISSUE TITLE
Unable to install razorpay-capacitor

### ISSUE THREAD

**Issue by iamromec** [razorpay/razorpay-capacitor#2]
It's stuck at 

<img width="706" alt="Screenshot 2021-03-16 at 10 37 41 PM" src="https://user-images.githubusercontent.com/50284628/111351353-1b447b80-86a9-11eb-8ff4-a02a90c5c777.png">

Not able to install this plugin. Could you please check if it's working fine? Tried several times.

**Comment by Nautiyalsachin** [comment]
@iamromec Can you try again? Let me know if you are still not able to implement it. Thanks.

**Comment by iamromec** [comment]
It did install but not working correctly on both platforms.

On iOS, no action is being done when I click on button and call 

`try {
      const data = (await Checkout.open(options));
      successCallback(data.response);
    } catch (error) {
      errorCallback(error.message);
    }`

On Android, once the payment is done, it does not call successCallback but crashes the Razorpay window without any response.

`2021-03-17 17:12:37.737 22213-22213/co.grocify E/ActivityThread: Activity com.razorpay.CheckoutActivity has leaked IntentReceiver com.razorpay.AutoReadOtpHelper@73e95eb that was originally registered here. Are you missing a call to unregisterReceiver()?
    android.app.IntentReceiverLeaked: Activity com.razorpay.CheckoutActivity has leaked IntentReceiver com.razorpay.AutoReadOtpHelper@73e95eb that was originally registered here. Are you missing a call to unregisterReceiver()?
        at android.app.LoadedApk$ReceiverDispatcher.<init>(LoadedApk.java:1603)
        at android.app.LoadedApk.getReceiverDispatcher(LoadedApk.java:1368)
        at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1539)
        at android.app.ContextImpl.registerReceiver(ContextImpl.java:1512)
        at android.app.ContextImpl.registerReceiver(ContextImpl.java:1500)
        at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:627)
        at com.razorpay.CheckoutPresenterImpl.onSubmit(CheckoutPresenterImpl.java:922)
        at com.razorpay.CheckoutBridge$10.secure(CheckoutBridge.java:157)
        at com.razorpay.CheckoutPresenterImpl.executeWebViewCallback(CheckoutPresenterImpl.java:1221)
        at com.razorpay.CheckoutPresenterImpl.access$700(CheckoutPresenterImpl.java:49)
        at com.razorpay.CheckoutPresenterImpl$17.run(CheckoutPresenterImpl.java:1207)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)`

`2021-03-17 17:12:37.738 22213-22213/co.grocify E/ActivityThread: Activity com.razorpay.CheckoutActivity has leaked IntentReceiver com.razorpay.CheckoutPresenterImpl$1@c6a16a9 that was originally registered here. Are you missing a call to unregisterReceiver()?
    android.app.IntentReceiverLeaked: Activity com.razorpay.CheckoutActivity has leaked IntentReceiver com.razorpay.CheckoutPresenterImpl$1@c6a16a9 that was originally registered here. Are you missing a call to unregisterReceiver()?
        at android.app.LoadedApk$ReceiverDispatcher.<init>(LoadedApk.java:1603)
        at android.app.LoadedApk.getReceiverDispatcher(LoadedApk.java:1368)
        at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1539)
        at android.app.ContextImpl.registerReceiver(ContextImpl.java:1512)
        at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:640)
        at com.razorpay.CheckoutPresenterImpl$8.onFinish(CheckoutPresenterImpl.java:764)
        at android.os.CountDownTimer$1.handleMessage(CountDownTimer.java:127)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
2021-03-17 17:12:42.348 22213-22722/co.grocify V/FA: Inactivity, disconnecting from the service
`

**Comment by iamromec** [comment]
I did pass the same options correctly...here 

`{
  "description": "Order Payment Transaction",
  "currency": "INR",
  "key": "raz_jey",
  "amount": 7290,
  "name": "Growcify",
  "prefill": {
    "email": "email@example.com",
    "contact": "80******38",
    "name": "IAMROMEC"
  },
  "theme": {
    "color": "#007bffFF"
  },
  "modal": {},
  "handler": {},
  "image": "image_url"
}`

**Comment by iamromec** [comment]
Hey @Nautiyalsachin, any update about the issue?

**Comment by iamromec** [comment]
Hey @Nautiyalsachin, is there any update, please? We are stand by on this. It's very important and urgent for us.

**Comment by ketankatdare** [comment]
Hey @Nautiyalsachin Same issue is encountered to me also. 
On the web it executes correctly returning the proper response.
On iOS it returns the empty response. And in android it just crashes like @iamromec mentioned.

**Comment by ketankatdare** [comment]
Hey @iamromec please checkout #5. Returning the response on android and ios.

**Comment by iamromec** [comment]
Waiting for this to be merged if it fixes the issue.

**Comment by Nautiyalsachin** [comment]
@iamromec: Please share a sample app and steps you have followed to integrate the capacitor plugin, also share the other projects details like your XCode version, pod version, etc.

@ketankatdare: The above PR is only related to fetch the data in callbacks, I can't see much changes here which can remove the above errors.

**Comment by iamromec** [comment]
@Nautiyalsachin 

Here's the link to sample app using Ionic Framework https://github.com/iamromec/razorpay-capacitor-sample-app

Sample app info
<img width="664" alt="Screenshot 2021-03-22 at 7 16 51 PM" src="https://user-images.githubusercontent.com/50284628/111999792-503a4d80-8b43-11eb-8371-53ede59ac6ad.png">

Sample app's package.json

```json
{
  "name": "razorpay-capacitor-sample-app",
  "version": "0.0.1",
  "author": "Ionic Framework",
  "homepage": "https://ionicframework.com/",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~11.2.0",
    "@angular/core": "~11.2.0",
    "@angular/forms": "~11.2.0",
    "@angular/platform-browser": "~11.2.0",
    "@angular/platform-browser-dynamic": "~11.2.0",
    "@angular/router": "~11.2.0",
    "@capacitor/android": "^2.4.7",
    "@capacitor/core": "2.4.7",
    "@capacitor/ios": "^2.4.7",
    "@ionic/angular": "^5.5.2",
    "capacitor-razorpay": "git+https://github.com/razorpay/razorpay-capacitor.git",
    "rxjs": "~6.6.0",
    "tslib": "^2.0.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.1102.4",
    "@angular/cli": "~11.2.4",
    "@angular/compiler": "~11.2.0",
    "@angular/compiler-cli": "~11.2.0",
    "@angular/language-service": "~11.2.0",
    "@capacitor/cli": "2.4.7",
    "@ionic/angular-toolkit": "^3.1.1",
    "@types/jasmine": "~3.6.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^12.11.1",
    "codelyzer": "^6.0.0",
    "jasmine-core": "~3.6.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~5.2.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~4.0.2"
  },
  "description": "An Ionic project"
}
```

My app's info
<img width="593" alt="Screenshot 2021-03-22 at 7 18 47 PM" src="https://user-images.githubusercontent.com/50284628/112000767-4a913780-8b44-11eb-86bd-1702a872bab4.png">

My app's package.json

```json
{
  "name": "app",
  "version": "0.0.1",
  "author": "Ionic Framework",
  "homepage": "https://ionicframework.com/",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "resources": "cordova-res ios && cordova-res android && node scripts/resources.js"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~9.1.6",
    "@angular/core": "~9.1.6",
    "@angular/forms": "~9.1.6",
    "@angular/platform-browser": "~9.1.6",
    "@angular/platform-browser-dynamic": "~9.1.6",
    "@angular/router": "~9.1.6",
    "@capacitor/android": "^2.4.0",
    "@capacitor/core": "2.4.0",
    "@capacitor/ios": "^2.4.0",
    "@ionic-native/camera": "^5.30.0",
    "@ionic-native/clipboard": "^5.30.0",
    "@ionic-native/core": "^5.22.0-beta-1",
    "@ionic-native/device": "^5.28.0",
    "@ionic-native/file": "^5.30.0",
    "@ionic-native/file-path": "^5.30.0",
    "@ionic-native/file-transfer": "^5.30.0",
    "@ionic-native/firebase-analytics": "^5.30.0",
    "@ionic-native/firebase-messaging": "^5.30.0",
    "@ionic-native/geolocation": "^5.27.0",
    "@ionic-native/google-maps": "^5.27.0-beta-20200630",
    "@ionic-native/http": "^5.27.0",
    "@ionic-native/in-app-browser": "^5.30.0",
    "@ionic-native/location-accuracy": "^5.30.0",
    "@ionic-native/social-sharing": "^5.30.0",
    "@ionic-native/speech-recognition": "^5.27.0",
    "@ionic-native/splash-screen": "^5.27.0",
    "@ionic-native/status-bar": "^5.27.0",
    "@ionic-native/toast": "^5.28.0",
    "@ionic/angular": "^5.0.0",
    "@ionic/storage": "^2.3.0",
    "@ngx-translate/core": "^13.0.0",
    "@ngx-translate/http-loader": "^6.0.0",
    "capacitor-razorpay": "git+https://github.com/razorpay/razorpay-capacitor.git",
    "cordova-androidx-build": "^1.0.4",
    "cordova-clipboard": "^1.3.0",
    "cordova-plugin-androidx": "^3.0.0",
    "cordova-plugin-camera": "^5.0.1",
    "cordova-plugin-cocoapod-support": "^1.6.2",
    "cordova-plugin-device": "^2.0.3",
    "cordova-plugin-file": "^6.0.2",
    "cordova-plugin-file-transfer": "^1.7.1",
    "cordova-plugin-filepath": "^1.5.8",
    "cordova-plugin-firebase-analytics": "^5.0.0",
    "cordova-plugin-firebase-messaging": "^5.0.0",
    "cordova-plugin-geolocation": "^4.0.2",
    "cordova-plugin-googlemaps": "git+https://github.com/mapsplugin/cordova-plugin-googlemaps.git#multiple_maps",
    "cordova-plugin-inappbrowser": "^4.1.0",
    "cordova-plugin-request-location-accuracy": "^2.3.0",
    "cordova-plugin-speechrecognition": "^1.2.0",
    "cordova-plugin-splashscreen": "^6.0.0",
    "cordova-plugin-statusbar": "^2.4.3",
    "cordova-plugin-x-socialsharing": "^6.0.2",
    "cordova-plugin-x-toast": "^2.7.2",
    "cordova-support-android-plugin": "^1.0.2",
    "es6-promise-plugin": "^4.2.2",
    "moment": "^2.27.0",
    "ngx-pinch-zoom": "^2.5.4",
    "qrcode": "^1.4.4",
    "rxjs": "~6.5.1",
    "tslib": "^1.10.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.901.5",
    "@angular/cli": "9.1.12",
    "@angular/compiler": "~9.1.6",
    "@angular/compiler-cli": "~9.1.6",
    "@angular/language-service": "~9.1.6",
    "@capacitor/cli": "2.4.0",
    "@ionic/angular-toolkit": "^2.1.1",
    "@ionic/lab": "3.1.7",
    "@types/googlemaps": "^3.39.13",
    "@types/jasmine": "~3.5.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^12.11.1",
    "codelyzer": "^5.1.2",
    "cordova-res": "^0.15.1",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~5.0.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~2.1.0",
    "karma-jasmine": "~3.0.1",
    "karma-jasmine-html-reporter": "^1.4.2",
    "protractor": "~5.4.3",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~3.8.3"
  },
  "description": "An Ionic project"
}
```

-----

Razorpay Capacitor is working ok with the sample app but not with our real app. And I want to use it with the ny app not the sample one.

-----
Steps I've used to install the Razorpay Capacitor plugin : 

`npm i -S https://github.com/razorpay/razorpay-capacitor.git`

That's it!

And then ran command to build Android & iOS.

-----
XCode Version: Version 12.4
Pod Version: None. I have not installed any razorpay pod separately



Let me know if you need anything else.

**Comment by iamromec** [comment]
Attaching the error of sample app's Android version too


<img width="1552" alt="Screenshot 2021-03-22 at 5 55 31 PM" src="https://user-images.githubusercontent.com/50284628/112002970-74e3f480-8b46-11eb-8bfe-f97f201787c9.png">

**Comment by vivekshindhe** [comment]
@iamromec @ketankatdare we've solved the issues with this. Waiting for merge and release. I will update here once done. Thanks for supporting us through this.

**Comment by iamromec** [comment]
@vivekagarwal1984 Will it fix for iOS too?

**Comment by iamromec** [comment]
@Nautiyalsachin @vivekshindhe any update? PLEASE TAKE THIS SERIOUSLY! Our business is being affected!

**Comment by vivekshindhe** [comment]
Its been updated. You can check it out by reinstalling the plugin. @iamromec

**Comment by iamromec** [comment]
No, it's not working! Again the same problem. It's stuck at installation step.

<img width="723" alt="Screenshot 2021-03-26 at 2 38 51 PM" src="https://user-images.githubusercontent.com/50284628/112609063-16b65a80-8e41-11eb-994b-24301449a08e.png">

It doesn't seem to be production ready.

**Comment by iamromec** [comment]
It did install but took half an hour. Will update whether it's working fine now or not.

**Comment by iamromec** [comment]
Still the issue persists!

**On Android, Net Banking isn't working nor any wallet processing. Only UPI payments are working fine.**

I've attached the screenshots...as there is no error response this time but not working.

<img width="1552" alt="Screenshot 2021-03-26 at 3 43 24 PM" src="https://user-images.githubusercontent.com/50284628/112618173-dd371c80-8e4b-11eb-9cbd-e4a69f441ce7.png">

Here's the app ui screen while making payment via Net Banking...nothing after this page!

![WhatsApp Image 2021-03-26 at 3 51 50 PM](https://user-images.githubusercontent.com/50284628/112618649-5a629180-8e4c-11eb-89e3-8a37226ca02d.jpeg)

**On iOS, still the same...no response at all!**

**Comment by iamromec** [comment]
FYI, I have re-installed the wrapper, added platforms again and then tried.

**Comment by Nautiyalsachin** [comment]
@iamromec Working on it, will update you on Tuesday, Thanks.

**Comment by vivekshindhe** [comment]
@iamromec I tried with the project you had shared with us.. I had to reinstall the ios platform after installing the plugin again, and run these commands, 
- ionic build
- npx cap add ios
- npx cap open ios
This is a still from the project you had shared with us, 
![Simulator Screen Shot - iPhone 12 mini - 2021-03-26 at 18 51 21](https://user-images.githubusercontent.com/57621906/112637549-4b87d900-8e64-11eb-9107-201d69991256.png)
Could you please try this once and get back to us on the same?

**Comment by iamromec** [comment]
@vivekshindhe With the sample app, it's showing fine in the iOS too. But obviously our production app is not just a sample app that only includes Razorpay Capacitor plugin. 

It's not working with the my package.json which I've shared above. It's does not work with that.

Here's the pay function's code:

`import 'capacitor-razorpay';
import { Plugins } from '@capacitor/core';
const { Checkout } = Plugins;`

`async pay() {
    const currencyCode = 'INR';
    const options = {"description":"Order Payment Transaction","currency":"INR","key":"rzp_live_key","amount": 5000,"name":"Growcify","prefill":{"email":"email@mail.co","contact":"xcxxxxxx94","name":"Full Name"},"theme":{"color":"#007bffFF"},"modal":{},"handler":{},"image":"img_url"};

    const newthis = this;
    const successCallback = function(response) {
      alert('payment_id: ' + response.razorpay_payment_id)
      //var orderId = success.razorpay_order_id
      //var signature = success.razorpay_signature
      newthis.successCallback(response.razorpay_payment_id);
    }

    const errorCallback = function(error) {
      newthis.errorCallback(error);
    }

    console.log('about...')

    console.log(options);
    try {
      console.log('heyy....');
      console.log('Checkout....', Checkout);
      const data = (await Checkout.open(options));
      console.log(data);
      successCallback(data.response);
    } catch (error) {
      errorCallback(error.message);
    }
  }
`
Here's the Xcode response, no action after 'hey....'
When I click on the 'PAY & PLACE ORDER', this is the response I get...no further processing...

![image](https://user-images.githubusercontent.com/50284628/112646997-11bbd000-8e6e-11eb-9378-48159a9f6a62.png)

The important thing to notice here is Checkout is empty!
<img width="1532" alt="Screenshot 2021-03-26 at 8 15 46 PM" src="https://user-images.githubusercontent.com/50284628/112649333-5a748880-8e70-11eb-9216-2ac1aa860f7b.png">

**Comment by vivekshindhe** [comment]
@iamromec I just checked this too... Getting Checkout as empty doesn't affect the issue. I'll need some more details here because I'm not able to reproduce this issue at all. Maybe we can get on a call and sort this out?

**Comment by iamromec** [comment]
What time?

**Comment by iamromec** [comment]
@vivekshindhe ??

**Comment by iamromec** [comment]
@Nautiyalsachin "Working on it, will update you on Tuesday, Thanks."

Any update? It's already Thursday.

**Comment by vivekshindhe** [comment]
@iamromec Apologies. Under the weather. We can have the call today. Let us know a time when you can accomodate.

**Comment by iamromec** [comment]
@vivekshindhe How about 3 pm today or share whatever works for you post 3 pm.

**Comment by vivekshindhe** [comment]
@iamromec now?

**Comment by vivekshindhe** [comment]
Have sent you an email with the invite. Please join when you can. @iamromec

**Comment by iamromec** [comment]
Thanks, @vivekshindhe, and @Nautiyalsachin for getting on call to fix this.

This was fixed by removing modal and handler from the options object.
      
````json    
   modal: {
        ondismiss: function () {
          console.log('dismissed')
        }
      },
    handler: {}
````

Make sure your options object does not have modal and handler properties in the all-new 'razorpay-capacitor' plugin.

Thanks.

**Comment by iamromec** [comment]
@Nautiyalsachin Is there a way around to get rid of "Add the Checkout class to the ArrayList in the MainActivity class in {{projectDir}}/android/src/main/MainActivity.java"

**Comment by vivekshindhe** [comment]
No @iamromec. Without that the plugin won't work for android. This is a mandated step even if CapacitorJs documents.

**Comment by vivekshindhe** [comment]
Closing issue. Please respond here or create a new ticket in case you have any queries.

**Comment by mukul-myhq** [comment]
HI @vivekshindhe, having similar issue where I'm not getting the response from `Checkout.open` call, it performs all the actions, i.e I can choose the payment type and enter my details, and finally it will show me success on the razorpay Modal. But its neither sending the order details nor any kind of errors back to me. The promise of `Checkout.open` never resolves for me. 
Working fine in case of iOS.

Version info:
Capacitor: V3 and V4 both
Razorpay: V1.1.0  and V1.2.0 both
Ionic: V5
Angular: V11

any pointers / need more info from me?
